app/scripts/lib/crypto/hkdf.js

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  /* eslint-disable */
+  const sjcl = require('sjcl')
+
+  /**
+   * hkdf - The HMAC-based Key Derivation Function
+   * based on https://github.com/mozilla/node-hkdf
```

Comment:
The latest SJCL appears to have a HKDF implementation built in:

  http://bitwiseshiftleft.github.io/sjcl/doc/hkdf.js.html

Are we using an older version that doesn't have this natively?

Reply:
So does WebCrypto: https://github.com/diafygi/webcrypto-examples#hkdf-ctr---derivebits

Reply:
@rfk I looked into using the latest version of SJCL, but saw that it was pinned to version [ac0b3fe0](https://github.com/bitwiseshiftleft/sjcl#security-advisories) for a security reason. Is this still a valid security issue for us?

If not, the upgrade would be helpful because they also have some base32 encoding functions. Either way, I'll take a look at SJCL and webcrypto hdkf functions since the recoveryKeyId generated does not match what is generated in your python script.